doc/api/readline.md

```diff
@@ -502,7 +531,352 @@ Returns the real position of the cursor in relation to the input
 prompt + string. Long input (wrapping) strings, as well as multiple
 line prompts are included in the calculations.
 
-## `readline.clearLine(stream, dir[, callback])`
+## Promises API
+<!-- YAML
+added: REPLACEME
+-->
+
+### Class: `readlinePromises.Interface`
+<!-- YAML
+added: REPLACEME
+-->
+
+* Extends: {readline.InterfaceConstructor}
+
+Instances of the `readlinePromises.Interface` class are constructed using the
+`readlinePromises.createInterface()` method. Every instance is associated with a
+single `input` [Readable][] stream and a single `output` [Writable][] stream.
+The `output` stream is used to print prompts for user input that arrives on,
+and is read from, the `input` stream.
+
+#### `rl.question(query[, options])`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `query` {string} A statement or query to write to `output`, prepended to the
+  prompt.
+* `options` {Object}
+  * `signal` {AbortSignal} Optionally allows the `question()` to be canceled
+    using an `AbortController`.
+* Returns: {Promise} A promise that is resolved with the user's
+  input in response to the `query`.
+
+The `rl.question()` method displays the `query` by writing it to the `output`,
+waits for user input to be provided on `input`, then invokes the `callback`
+function passing the provided input as the first argument.
+
+When called, `rl.question()` will resume the `input` stream if it has been
+paused.
+
+If the `readlinePromises.Interface` was created with `output` set to `null` or
+`undefined` the `query` is not written.
+
+Example usage:
+
+```mjs
+const answer = await rl.question('What is your favorite food? ');
+console.log(`Oh, so your favorite food is ${answer}`);
+```
+
+Using an `AbortController` to cancel a question.
+
+```mjs
+const ac = new AbortController();
+const signal = ac.signal;
+
+const answer = await rl.question('What is your favorite food? ', { signal });
+console.log(`Oh, so your favorite food is ${answer}`);
+
+signal.addEventListener('abort', () => {
+  console.log('The food question timed out');
+}, { once: true });
+
+setTimeout(() => ac.abort(), 10000);
+```
+
+### `readlinePromises.Readline(stream)`
```

Comment:
This doesn't really follow the pattern we use for other classes in the docs. Generally we'd have something like `### Class: `readlinePromises.Readline`, and for the constructor something like `#### new readlinePromises.Readline(stream[, options])` ... Not blocking because the consistency can be worked on in a separate PR